spock-specs/src/test/groovy/org/spockframework/smoke/mock/OrderedInteractions.groovy

```diff
@@ -98,44 +165,54 @@ class OrderedInteractions extends Specification {
   }
 }
 
-class RemainingBehaviorSameAsForUnorderedInteractions extends Specification {
-  @FailsWith(TooFewInvocationsError)
+class RemainingBehaviorSameAsForUnorderedInteractions extends EmbeddedSpecification {
   def "too few invocations for one of the when-blocks"() {
-    def list = Mock(List)
-
     when:
-    list.add("foo")
-    list.add("bar")
+    runner.runFeatureBody """
```

Comment:
I don't think that is necessarily better, when we do no further validation on exception using `@FailsWith` is more succinct and it doesn't require that the code is put into a String. While EmbeddedSpecs are powerful, they add a layer of indirection, that is not necessary here.

Reply:
Well, I just did what the documentation says and what seems semantically meaningfuller, which is

> or as a replacement for
> exception conditions in certain corner cases where they cannot be used (like
> specifying the behavior of exception conditions). In all other cases,
> exception conditions are preferable.

But ok, changed it back in the three cases where I don't check the message. :-)

Reply:
IMO this counts as such a corner case, since you basically have to emulate the whole feature just so you can use the exception condition.